Device layout editor

xQwexx opened the pull request:
Enable multiple device controlling, it work in multiple layout handling
Adding device layout editor
But the physical device handling not ready yet, firstly the device loading from dlls needs to merge.

**List any issues that this PR fixes:** fixes # , etc...
Fixes: #1871 #1844 #1033 #1587 #1027
because it updates the text of the keys, the language layouts not needed anymore:
#16 #1251 #1422


**Known issues/To do:**

**Not compatible the appdata, so first save it before running this.**

xQwexx committed:
Dynamical layout json files scanning

xQwexx committed:
Minor code cleaning in the LayoutManager

xQwexx committed:
Refactor to enable multiple device layouts

xQwexx committed:
Enable to move deviceKeys and some fixes

xQwexx committed:
Merge branch 'dev' of https://github.com/antonpup/Aurora into feature/multiple-device-layout

# Conflicts:
#	Project-Aurora/Project-Aurora/App.xaml.cs
#	Project-Aurora/Project-Aurora/Controls/KeySequence.xaml.cs
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Corsair_strafe_mk2_logo.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Roccat_Kone_Pure_logo.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Roccat_Kone_Pure_outline.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/corsair_strafe_mk2.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/ducky_one_2_rgb_tkl.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/ducky_shine_7.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/masterkeys_mk730.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/corsair_strafe_mk2_left_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/corsair_strafe_mk2_right_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/ducky_shine_7_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/masterkeys_mk730_bottom_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/masterkeys_mk730_left_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/masterkeys_mk730_right_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/roccat_kone_pure_features.json
#	Project-Aurora/Project-Aurora/Devices/CoolerMaster/CoolerMasterDevice.cs
#	Project-Aurora/Project-Aurora/EffectsEngine/EffectLayer.cs
#	Project-Aurora/Project-Aurora/Profiles/CSGO/Layers/CSGOBombLayerHandler.cs
#	Project-Aurora/Project-Aurora/Profiles/CSGO/Layers/CSGOKillsIndicatorLayerHandler.cs
#	Project-Aurora/Project-Aurora/Profiles/LeagueOfLegends/LoLProfile.cs
#	Project-Aurora/Project-Aurora/Profiles/Minecraft/MinecraftProfile.cs
#	Project-Aurora/Project-Aurora/Project-Aurora.csproj
#	Project-Aurora/Project-Aurora/Settings/KeyboardLayoutManager.cs

xQwexx committed:
enable loading old profiles, Keycap control refactor

xQwexx committed:
Add control to edit DeviceKeycapConf
Save edited layouts

xQwexx committed:
Merge branch 'dev' of https://github.com/antonpup/Aurora into feature/multiple-device-layout

# Conflicts:
#	Project-Aurora/Project-Aurora/Profiles/Skype/SkypeProfile.cs
#	Project-Aurora/Project-Aurora/Project-Aurora.csproj

xQwexx committed:
Merge branch 'dev' into feature/multiple-device-layout

Bluscream commented:
O M G

xQwexx committed:
Copy json if it newer, delete unnecessary ones

Tyberious commented:
I am digging this, keep up the good work.  This should make it easier to add ram and fans etc once you refine the interface.

xQwexx committed:
Not found directory fix

xQwexx committed:
Merge branch 'dev' of https://github.com/antonpup/Aurora into feature/multiple-device-layout

# Conflicts:
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/Uniwill2ND_35X_1.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/Uniwill2ND_35X_2.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/Uniwill2P1_550_BR.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/Uniwill2P1_550_JP.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/Uniwill2P1_550_UK.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/Uniwill2P1_550_US.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/Uniwill2P2_650_BR.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/Uniwill2P2_650_JP.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/Uniwill2P2_650_UK.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/Uniwill2P2_650_US.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/keyboard101_numpad_right_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/keyboard102_numpad_right_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/keyboard21_numpad_left_bottom_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/keyboard21br_numpad_left_bottom_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/keyboard21jp_numpad_left_bottom_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/keyboard21us_numpad_right_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/keyboard22br_numpad_left_bottom_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/keyboard22jp_numpad_left_bottom_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/keyboard22us_numpad_right_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/keyboard22usuk_numpad_left_bottom_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/keyboard_numpad_left_bottom_features.json
#	Project-Aurora/Project-Aurora/Project-Aurora.csproj
#	Project-Aurora/Project-Aurora/Settings/KeyboardLayoutManager.cs

xQwexx committed:
Enable separate japan layout editing, convert physical layout Types to enum

xQwexx committed:
Merge branch 'dev' of https://github.com/antonpup/Aurora into feature/multiple-device-layout

# Conflicts:
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/logitech_g513.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/logitech_g815.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/logitech_g815_features.json
#	Project-Aurora/Project-Aurora/Devices/YeeLight/YeeLightDevice.cs
#	Project-Aurora/Project-Aurora/Project-Aurora.csproj
#	Project-Aurora/Project-Aurora/Settings/KeyboardLayoutManager.cs

xQwexx committed:
All current bug fix + Merge branch 'dev' 'Asus Change'

# Conflicts:
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Asus_logo.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/asus_strix_scope.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/asus_strix_flare_features.json
#	Project-Aurora/Project-Aurora/Project-Aurora.csproj
#	Project-Aurora/Project-Aurora/Settings/KeyboardLayoutManager.cs

xQwexx committed:
Merge branch 'dev' of https://github.com/antonpup/Aurora into feature/multiple-device-layout

xQwexx committed:
small fix

xQwexx committed:
Fix exception when device layout offset was negative

Bluscream commented:
Just wait with the merging from dev to your fork until the review is passed, otherwise you'll spend your whole life updating your fork :P

xQwexx committed:
Fix device selection, update visualname change, refactor deviceLayout

xQwexx committed:
Update no device selected box

xQwexx committed:
Fix layout not applying the colors

xQwexx committed:
Bug fixes, add keyboard physical layout detection, Fix bitmap lock exception

xQwexx committed:
Fix keyselection selection

xQwexx committed:
Optimizing the DevicePresenter, fix other bugs

xQwexx committed:
Resize freeform thumbs

xQwexx committed:
Merge branch 'dev' of https://github.com/xQwexx/Aurora into feature/multiple-device-layout

# Conflicts:
#	Project-Aurora/Project-Aurora/ConfigUI.xaml.cs
#	Project-Aurora/Project-Aurora/Controls/KeySequence.xaml
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Logi_G502_dpi.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Logi_g900_dpi.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Omen_zone_0.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Omen_zone_1.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Omen_zone_2.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Omen_zone_3.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE.psd
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_L1.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_L2.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_L3.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_L4.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_L5.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_L6.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_L7.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_Logo.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_R1.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_R2.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_R3.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_R4.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_R5.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_R6.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_R7.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_Scroll_Wheel.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/Razer_Mamba_TE_outline.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/omen_outpost_outline.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/omen_outpost_plus_photon_outline.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/omen_photon_outline.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/omen_photon_scrollwheel.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/omen_sequencer_logo.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/omen_vector_essentials_outline.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/omen_vector_logo.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Images/omen_vector_outline.png
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/cooler_master_sk650.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/hyperx_alloy_elite_rgb.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/omen_four_zone.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Keyboard/omen_sequencer.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/cooler_master_sk650_bottom_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/cooler_master_sk650_left_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/cooler_master_sk650_right_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/cooler_master_sk650_top_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/hyperx_alloy_elite_rgb_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/omen_four_zone_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/omen_outpost_plus_photon_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/omen_photon_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/omen_sequencer_bottom_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/omen_sequencer_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/omen_sequencer_right_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/omen_vector_essentials_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/omen_vector_features.json
#	Project-Aurora/Project-Aurora/DeviceLayouts/Mouse/razer_mamba_te_features.json
#	Project-Aurora/Project-Aurora/Devices/Asus/AsusDevice.cs
#	Project-Aurora/Project-Aurora/Devices/Asus/AuraSyncKeyboardDevice.cs
#	Project-Aurora/Project-Aurora/Devices/CoolerMaster/CoolerMasterDevice.cs
#	Project-Aurora/Project-Aurora/Devices/DeviceKeys.cs
#	Project-Aurora/Project-Aurora/Devices/DeviceManager.cs
#	Project-Aurora/Project-Aurora/Devices/Drevo/DrevoDevice.cs
#	Project-Aurora/Project-Aurora/Devices/Dualshock4/Dualshock4Device.cs
#	Project-Aurora/Project-Aurora/Devices/Logitech/LogitechDevice.cs
#	Project-Aurora/Project-Aurora/Devices/NZXT/NZXTDevice.cs
#	Project-Aurora/Project-Aurora/Devices/RGBNet/AbstractRGBNetDevice.cs
#	Project-Aurora/Project-Aurora/Devices/RGBNet/AuroraRGBNetBrush.cs
#	Project-Aurora/Project-Aurora/Devices/Razer/RazerDevice.cs
#	Project-Aurora/Project-Aurora/Devices/Vulcan/Vulcan.cs
#	Project-Aurora/Project-Aurora/Devices/Wooting/WootingDevice.cs
#	Project-Aurora/Project-Aurora/Devices/YeeLight/YeeLightDevice.cs
#	Project-Aurora/Project-Aurora/Profiles/ATS/Control_ATS.xaml
#	Project-Aurora/Project-Aurora/Profiles/Battlefield1/Control_Battlefield1.xaml
#	Project-Aurora/Project-Aurora/Profiles/Blade and Soul/Control_BnS.xaml
#	Project-Aurora/Project-Aurora/Profiles/CSGO/Control_CSGO.xaml
#	Project-Aurora/Project-Aurora/Profiles/DOOM/Control_DOOM.xaml
#	Project-Aurora/Project-Aurora/Profiles/Desktop/Control_Desktop.xaml
#	Project-Aurora/Project-Aurora/Profiles/DiscoDodgeball/Control_DiscoDodgeball.xaml
#	Project-Aurora/Project-Aurora/Profiles/Dota 2/Control_Dota2.xaml
#	Project-Aurora/Project-Aurora/Profiles/DyingLight/Control_DyingLight.xaml
#	Project-Aurora/Project-Aurora/Profiles/EliteDangerous/Control_EliteDangerous.xaml
#	Project-Aurora/Project-Aurora/Profiles/Factorio/Control_Factorio.xaml
#	Project-Aurora/Project-Aurora/Profiles/GTA5/Control_GTA5.xaml
#	Project-Aurora/Project-Aurora/Profiles/LeagueOfLegends/Control_LoL.xaml
#	Project-Aurora/Project-Aurora/Profiles/Move or Die/Control_MoD.xaml
#	Project-Aurora/Project-Aurora/Profiles/Overwatch/Control_Overwatch.xaml
#	Project-Aurora/Project-Aurora/Profiles/Payday 2/Control_PD2.xaml
#	Project-Aurora/Project-Aurora/Profiles/QuantumConumdrum/Control_QuantumConumdrum.xaml
#	Project-Aurora/Project-Aurora/Profiles/Serious Sam 3/Control_SSam3.xaml
#	Project-Aurora/Project-Aurora/Profiles/ShadowOfMordor/Control_ShadowOfMordor.xaml
#	Project-Aurora/Project-Aurora/Profiles/Witcher3/Control_Witcher3.xaml
#	Project-Aurora/Project-Aurora/Profiles/XCOM/Control_XCOM.xaml
#	Project-Aurora/Project-Aurora/Project-Aurora.csproj
#	Project-Aurora/Project-Aurora/Settings/Configuration.cs
#	Project-Aurora/Project-Aurora/Settings/Control_Settings.xaml
#	Project-Aurora/Project-Aurora/Settings/Control_Settings.xaml.cs
#	Project-Aurora/Project-Aurora/Settings/KeyboardLayoutManager.cs
#	Project-Aurora/Project-Aurora/Settings/Layers/Control_RazerLayer.xaml
#	Project-Aurora/Project-Aurora/Utils/EnumUtils.cs
#	Project-Aurora/Project-Aurora/kb_layouts/Extra Features/logitech_g502_features.json
#	Project-Aurora/Project-Aurora/kb_layouts/Extra Features/logitech_g900_features.json

xQwexx committed:
Add OpenRGB multi device support

xQwexx committed:
Add Cointainer Wrapper, Fix 0 index for Keyboard, and other bug fixes

xQwexx committed:
Fix async not working proberly, One openRGB device just one layout

xQwexx committed:
Fix for #2320

xQwexx committed:
Add invisible background json flag and some performance optimalization

xQwexx committed:
Fix selection Key Layout, finalizing the Control_Keys

xQwexx committed:
Small editor improvement

xQwexx committed:
Add Steelseries rival 650 layout and disable keycap dragging

xQwexx committed:
The custom layout stores in the appdata

xQwexx committed:
Enable Keycap dragging

xQwexx committed:
Replace corsair sdk support

xQwexx committed:
Add not existing keyboard keys

xQwexx committed:
Fix Razer Sdk init fail, missed index

xQwexx committed:
Fix custom device layouts are not displayed in the drop down

xQwexx committed:
Fix old way of serrializate keyboard

xQwexx committed:
Add +/- and auto devicename discovery

xQwexx committed:
Set active layoutname after you save it

xQwexx committed:
Add corsair sdk suppport

xQwexx committed:
Remove CorairRGB.NET dependency

xQwexx committed:
Fix update flash not happening

xQwexx committed:
Fix corsair detection

xQwexx committed:
Fix ghosting device layout bug

xQwexx committed:
Fix Corsair key mapping

xQwexx committed:
Fix some device layout bugs

xQwexx committed:
Update Colore version

boehs commented:
I hope @diogotr7 considers merging, this is very nice, and goodness knows we need a big of a refactor to spark some light in this project

xQwexx committed:
Fix helping text
